tests/unit/Rules/OddTest.php

```diff
@@ -13,60 +13,47 @@
 
 namespace Respect\Validation\Rules;
 
-use PHPUnit\Framework\TestCase;
+use Respect\Validation\Test\RuleTestCase;
 
 /**
- * @group  rule
+ * @group rule
  * @covers \Respect\Validation\Rules\Odd
- * @covers \Respect\Validation\Exceptions\OddException
+ *
+ * @author Danilo Benevides <danilobenevides01@gmail.com>
+ * @author Gabriel Caruso <carusogabriel34@gmail.com>
+ * @author Henrique Moody <henriquemoody@gmail.com>
+ * @author Jean Pimentel <jeanfap@gmail.com>
  */
-class OddTest extends TestCase
+final class OddTest extends RuleTestCase
 {
-    protected $object;
-
-    protected function setUp(): void
-    {
-        $this->object = new Odd();
-    }
-
-    /**
-     * @dataProvider providerForOdd
-     */
-    public function testOdd($input): void
+    /*
+    * {@inheritdoc}
+    */
+    public function providerForValidInput(): array
     {
-        $this->object->assert($input);
-        self::assertTrue($this->object->__invoke($input));
-        $this->object->check($input);
-    }
+        $rule = new Odd();
 
-    /**
-     * @dataProvider providerForNotOdd
-     * @expectedException \Respect\Validation\Exceptions\OddException
-     */
-    public function testNotOdd($input): void
-    {
-        self::assertFalse($this->object->__invoke($input));
-        $this->object->assert($input);
-    }
-
-    public function providerForOdd()
-    {
         return [
-            [-5],
-            [-1],
-            [1],
-            [13],
+            [$rule, -5],
+            [$rule, -1],
+            [$rule, 1],
+            [$rule, 13],
         ];
     }
 
-    public function providerForNotOdd()
+    /*
+    * {@inheritdoc}
+    */
+    public function providerForInvalidInput(): array
     {
+        $rule = new Odd();
+
         return [
-            [''],
-            [-2],
-            [-0],
-            [0],
-            [32],
+            [$rule, ''],
```

Comment:
Because this PR is adding a new behaviour it should have added a few more tests, and updates the changelog of the rule.